pkg/scheduler/framework/runtime/framework.go

```diff
@@ -415,13 +425,22 @@ func registerClusterEvents(name string, eventToPlugins map[framework.ClusterEven
 // a default from the scheme. Returns `nil, nil` if the plugin does not have a
 // defined arg types, such as in-tree plugins that don't require configuration
 // or out-of-tree plugins.
-func getPluginArgsOrDefault(pluginConfig map[string]runtime.Object, name string) (runtime.Object, error) {
+func getPluginArgsOrDefault(componentConfigVersion string, pluginConfig map[string]runtime.Object, name string) (runtime.Object, error) {
 	res, ok := pluginConfig[name]
 	if ok {
 		return res, nil
 	}
 	// Use defaults from latest config API version.
-	gvk := v1beta1.SchemeGroupVersion.WithKind(name + "Args")
+	var gvk schema.GroupVersionKind
+	switch componentConfigVersion {
+	case v1beta1.SchemeGroupVersion.String():
+		gvk = v1beta1.SchemeGroupVersion.WithKind(name + "Args")
+	case v1beta2.SchemeGroupVersion.String():
+		gvk = v1beta2.SchemeGroupVersion.WithKind(name + "Args")
+	default:
```

Comment:
should we default to this even if component config version is non-empty and is an unknown version?

Reply:
The original code was doing it regardless of `componentConfigVersion`, I think this is fine as is. In any case, this defaulting logic will be removed as a followup as part of https://github.com/kubernetes/kubernetes/issues/102211